src/dav.js

```diff
@@ -0,0 +1,376 @@
+      }
+    }
+    body += '  </d:prop>\n'
+    body += '</d:propfind>'
+
+    return this.request('PROPFIND', path, headers, body, options).then(
+      function (result) {
+        if (depth === '0') {
+          return {
+            status: result.status,
+            body: result.body[0],
+            res: result
+          }
+        } else {
+          return {
+            status: result.status,
+            body: result.body,
+            res: result
+          }
+        }
+      }
+    )
+  }
+
+  /**
+   * Renders a "d:set" block for the given properties.
+   *
+   * @param {Object.<String,String>} properties
+   * @return {String} XML "<d:set>" block
+   */
+  _renderPropSet (properties) {
+    var body = '  <d:set>\n' +
+          '   <d:prop>\n'
+
+    for (var ii in properties) {
+      // eslint-disable-next-line no-prototype-builtins
+      if (!properties.hasOwnProperty(ii)) {
+        continue
+      }
+
+      var property = this.parseClarkNotation(ii)
+      var propName
+      var propValue = properties[ii]
+      if (this.xmlNamespaces[property.namespace]) {
+        propName = this.xmlNamespaces[property.namespace] + ':' + property.name
+      } else {
+        propName = 'x:' + property.name + ' xmlns:x="' + property.namespace + '"'
+      }
+
+      // FIXME: hard-coded for now until we allow properties to
+      // specify whether to be escaped or not
+      if (propName !== 'd:resourcetype') {
+        propValue = this._escapeXml(propValue)
+      }
+      body += '      <' + propName + '>' + propValue + '</' + propName + '>\n'
+    }
+    body += '    </d:prop>\n'
+    body += '  </d:set>\n'
+    return body
+  }
+
+  /**
+   * Generates a propPatch request.
+   *
+   * @param {string} url Url to do the proppatch request on
+   * @param {Object.<String,String>} properties List of properties to store.
+   * @param {Object} [headers] headers
+   * @return {Promise}
+   */
+  propPatch (url, properties, headers) {
+    headers = headers || {}
+
+    headers['Content-Type'] = 'application/xml; charset=utf-8'
+
+    var body =
+          '<?xml version="1.0"?>\n' +
+          '<d:propertyupdate '
+    var namespace
+    for (namespace in this.xmlNamespaces) {
+      body += ' xmlns:' + this.xmlNamespaces[namespace] + '="' + namespace + '"'
+    }
+    body += '>\n' + this._renderPropSet(properties)
+    body += '</d:propertyupdate>'
+
+    return this.request('PROPPATCH', url, headers, body).then(
+      function (result) {
+        return {
+          status: result.status,
+          body: result.body,
+          res: result
+        }
+      }
+    )
+  }
+
+  /**
+   * Generates a MKCOL request.
+   * If attributes are given, it will use an extended MKCOL request.
+   *
+   * @param {string} url Url to do the proppatch request on
+   * @param {Object.<String,String>} [properties] list of properties to store.
+   * @param {Object} [headers] headers
+   * @return {Promise}
+   */
+  mkcol (url, properties, headers) {
```

Comment:
unit tests for this function would be great, to make sure the composed body is correct.

Reply:
I think that is covered very well by the pact tests. These methods make HTTP requests if we want to do unit tests we will need to intercept the HTTP request and check but not really necessary IMO